src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -131,14 +131,51 @@ public NamedPipeTest_CurrentUserOnly_Windows(TestAccountImpersonator testAccount
             _testAccountImpersonator = testAccountImpersonator;
         }
 
+        [OuterLoop]
+        [ConditionalFact(nameof(IsAdminOnSupportedWindowsVersions))]
+        public async Task Connection_UnderDifferentUsers_CurrentUserOnlyOnServer_DoesNotBlockOthers()
+        {
+            string name = PipeStreamConformanceTests.GetUniquePipeName();
+            bool invalidClientShouldStop = false;
+
+            using var server = new NamedPipeServerStream(name, PipeDirection.InOut, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous | PipeOptions.CurrentUserOnly);
+            Task serverWait = server.WaitForConnectionAsync();
+
+            Task invalidClient = Task.Run(() =>
+            {
+                // invalid non-current user tries to connect to server.
+                _testAccountImpersonator.RunImpersonated(() =>
```

Comment:
Noting here that I believe this user will not have admin privileges, which I think is what we want. You might consider adding an assert to document that eg
`Assert.False(currentIdentity.IsInRole(WindowsBuiltInRole.Administrator));`  after line 106.